doc/users/whats_new/path_simplification_updates.rst

```diff
@@ -0,0 +1,8 @@
+Path simplification updates
+---------------------------
+
+Lines will now be simplified according to the ``path.simplify`` and
+``path.simplify_threshold`` RC parameters, as long as other conditions
+such as not using dashed lines are met. Simplification will only happen
+when drawing the line segment portion of a path, if you are also drawing
+markers you should consider using the ``markevery`` option to ``plot``.
```

Comment:
Yeah I think it would be cleaner as a separate PR (I'm willing to do it, but I'll be focusing on this and other performance optimizations first). @anntzer @tacaswell the idea was that `path.simplify` could be removed because it is redundant with `path.simplify_threshold == 0`.